### ISSUE TITLE
Use an intermediate language

### ISSUE THREAD

**Issue by JeanJPNM** [mlogjs/mlogjs#79]
This is more of a suggestion than an actual request, using an intermediate language that doesn't have as many edge cases can make optimizations more precise.

**Comment by weisrc** [comment]
Indeed, this is what I planned to do https://github.com/mlogjs/optimlog. However, instead of creating some sort of intermediate language, I am just using plain MLog as the IR with a lot of metadata getting inferred from it.
